cmd/hdfs/put.go

```diff
@@ -29,6 +31,48 @@ func put(args []string) {
 		fatal(err)
 	}
 
+	if filepath.Base(source) == "-" {
+		putFromStdIn(client, dest)
+	} else {
+		putFromFile(client, source, dest)
+	}
+}
+
+func putFromStdIn(client *hdfs.Client, dest string) {
+	// If the destination exists, regardless of what it is, bail out.
+	existing, err := client.Stat(dest)
+	if err == nil {
+		if existing.IsDir() {
+			fatal(fmt.Errorf("can't write from STDIN into a directory: %s", dest))
+		} else {
+			fatal(fmt.Errorf("won't write to an existing file: %s", dest))
+		}
+	} else if !os.IsNotExist(err) {
+		fatal(err)
+	}
+
+	mode := 0755 | os.ModeDir
+	// make sure parent dir exists
```

Comment:
This and the next few comments are unnecessary. Also, for future reference, comments should be in complete sentences.